lms/djangoapps/instructor/views/api.py

```diff
@@ -1151,15 +1184,32 @@ def generate_registration_codes(request, course_id):
         internal_reference=internal_reference,
         customer_reference_number=customer_reference_number
     )
+
+    course = get_course_by_id(course_id, depth=None)
+    paid_modes = [mode for mode in CourseMode.modes_for_course(course_id) if mode.min_price > 0]
```

Comment:
I'd prefer if this were encapsulated in the `CourseMode` class.  Maybe add a kwarg to `modes_for_course` that filters out non-paid courses?